From 8299fe0615638dc88694f08b95a62e9dd3943a6f Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Date: Wed, 10 Dec 2025 02:27:16 +0200
Subject: [PATCH] freedreno: don't encode build path into binaries

Encoding build-specific path into installed binaries is generally
frowned upon. It harms the reproducibility of the build and e.g.
OpenEmbedded now considers that to be an error.

Instead of hardcoding rnn_src_path into the RNN_DEF_PATH, use the
relative path when running the build / test.

Additionaly make sure that RNN_PATH is set correctly when running `meson
devenv` in order to allow developers to call tools without installing
XML files to the target system.

Suggested-by: Rob Clark <rob.clark@oss.qualcomm.com>
Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Upstream-Status: Submitted [https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/30206]
[locally generated patch after git am'ing it to remove fuzz]
Signed-off-by: Quentin Schulz <quentin.schulz@cherry.de>
---
 src/freedreno/meson.build           | 5 +++--
 src/freedreno/registers/meson.build | 2 ++
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/freedreno/meson.build b/src/freedreno/meson.build
index a4bfd337179..330b6468d01 100644
--- a/src/freedreno/meson.build
+++ b/src/freedreno/meson.build
@@ -4,9 +4,10 @@
 inc_freedreno = include_directories(['.', './registers', './registers/adreno', './common'])
 inc_freedreno_rnn = include_directories('rnn')
 
-rnn_src_path = dir_source_root + '/src/freedreno/registers'
+rnn_rel_path = 'src/freedreno/registers'
+rnn_src_path = dir_source_root + '/' + rnn_rel_path
 rnn_install_path = get_option('datadir') + '/freedreno/registers'
-rnn_path = rnn_src_path + ':' + get_option('prefix') + '/' + rnn_install_path
+rnn_path = rnn_rel_path + ':' + get_option('prefix') + '/' + rnn_install_path
 
 dep_libarchive = dependency('libarchive', allow_fallback: true, required: false, disabler : true)
 dep_libxml2 = dependency('libxml-2.0', allow_fallback: true, required: false)
diff --git a/src/freedreno/registers/meson.build b/src/freedreno/registers/meson.build
index f4be68117c3..cd0caafa431 100644
--- a/src/freedreno/registers/meson.build
+++ b/src/freedreno/registers/meson.build
@@ -43,4 +43,6 @@ if install_fd_decode_tools
   endforeach
 endif
 
+devenv.set('RNN_PATH', meson.current_source_dir())
+
 subdir('adreno')
